lib/public/Util.php

```diff
@@ -173,10 +176,71 @@ public static function addStyle($application, $file = null) {
 	 * add a javascript file
 	 * @param string $application
 	 * @param string $file
+	 * @param string $afterAppId
 	 * @since 4.0.0
 	 */
-	public static function addScript($application, $file = null) {
-		\OC_Util::addScript($application, $file);
+	public static function addScript($application, $file = null, $afterAppId = null) {
+		if (!empty($application)) {
+			$path = "$application/js/$file";
+		} else {
+			$path = "js/$file";
+		}
+
+		// Inject js translations if we load a script for
+		// a specific app that is not core, as those js files
+		// need separate handling
+		if ($application !== 'core' && $file !== null) {
+			self::addTranslations($application);
+		}
+
+		// init app array if it doesn't exists
+		if (!array_key_exists($afterAppId, self::$scripts)) {
+			self::$scripts[$afterAppId] = ['first' => [], 'last' => []];
+		}
+
+		/**
+		 * manage priorities if defined
+		 * we store the data like this, then flatten everything
+		 * [
+		 * 'core' => [
+		 * 'first' => [
+		 * '/core/js/main.js',
+		 * ],
+		 * 'last' => [
+		 * '/apps/viewer/js/viewer-main.js',
+		 * ]
+		 * ],
+		 * 'viewer' => [
+		 * 'first' => [
+		 * '/apps/viewer/js/viewer-public.js',
+		 * ],
+		 * 'last' => [
+		 * '/apps/files_pdfviewer/js/files_pdfviewer-main.js',
+		 * ]
+		 * ]
+		 * ]
+		 */
+		if (!empty($afterAppId)) {
```

Comment:
Is empty is not checking for null, right?

https://devdocs.io/php/function.empty

```suggestion
 		if ($afterAppId !== null) {
```

Reply:
It does :dancing_women: